code/game/objects/items/frames/table_rack.dm

```diff
@@ -17,11 +17,13 @@
 	attack_verb = list("slammed", "bashed", "battered", "bludgeoned", "thrashed", "whacked")
 	var/table_type = /obj/structure/table //what type of table it creates when assembled
 	var/deconstruct_type = /obj/item/stack/sheet/metal
+	//do we drop metal when deconstructing parts?
+	var/dropmetal_on_deconstruct = TRUE
```

Comment:
couldnt you just deconstruct_type = null on the child and check if deconstruct_type exists instead?